File: lang/python/src/muse/algorithms/number_theory/LeastCommonMultiple.py
from muse.algorithms.number_theory import GreatestCommonDivisor

def reduceToBinaryGCD(m: int, n: int) -> int:
    if m < 0:
        m = -m
    if n < 0:
        n = -n

    if m == 0 or n == 0:
        return 0
    return m // GreatestCommonDivisor.iterativeBinaryGCD(m, n) * n

def reduceToEuclidean(m: int, n: int) -> int:
    if m < 0:
        m = -m
    if n < 0:
        n = -n

    if m == 0 or n == 0:
        return 0
    return m // GreatestCommonDivisor.iterativeEuclidean(m, n) * n

File: go/muse/algorithms/number_theory/least_common_multiple/least_common_multiple.go
package least_common_multiple

import "muse/algorithms/number_theory/greatest_common_divisor"

func ReduceToBinaryGCD(m, n int64) int64 {
	if m < 0 {
		m = -m
	}
	if n < 0 {
		n = -n
	}

	if m == 0 || n == 0 {
		return 0
	}
	return m / greatest_common_divisor.IterativeBinaryGCD(m, n) * n
}

func ReduceToEuclidean(m, n int64) int64 {
	if m < 0 {
		m = -m
	}
	if n < 0 {
		n = -n
	}

	if m == 0 || n == 0 {
		return 0
	}
	return m / greatest_common_divisor.IterativeEuclidean(m, n) * n
}
